### PR TITLE
Add acl_safe processing to JSONFragment

### PR DESCRIPTION
**Pull request by IvLaptev**
Added processing of `acl_safe` flag with JSONFragment. The process is similar to partial generators. JSONFragment generators has `acl_safe` method now. It returns ACL rules as JSON Pointer. While generating configuration generator either `acl()` or `acl_safe()`.

### CODE REVIEW

**Comment by azryve** [comment]
Sorry but I have to question this change. The `JSONFragment` generators are already implement is_safe() API which behave exactly as Entire.is_safe(). So the questions are

1) Do we really need to have both `Partial` and `Entire` approaches in `JSONFragment` ?
2) If not, then why we want the partial one, instead of splitting the big `JSONFragment` into two safe and unsafe ones

**Comment by IvLaptev** [comment]
In my opinion approaches were mixed earlier by someone else. There is no `acl()` method in entire generators, but it appears in `JSONFragment`. Also I can't find any mentions of overridden `is_safe()` in `JSONFragment`.

I think that we don't need both approaches. In my mind we should use the partial one because it is more flexible than the other. For example, we have `JSONFragment` generators with up to 10 ACL rules now. Some of that parts of config can be safe while others are unsafe. In this case we will be able remove some rules from `acl_safe()`, and config will be created correctly.

**Comment by IvLaptev** [comment]
Moreover, I think we can remove `is_safe()` from `JSONFragment` but I don't know if someone else is using it now.

File: annet/gen.py [hunk 1]
```diff
@@ -239,6 +239,7 @@ def _old_new_per_device(ctx: OldNewDeviceContext, device: Device, filterer: Filt
         new_files = res.new_files()
         new_json_fragment_files = res.new_json_fragment_files(old_json_fragment_files)
 
+        filters = None
```
Comment (azryve) [inline annet/gen.py:242]: `filters: List[str] = []` ?

**Comment by azryve** [comment]
I dont think that it is a mixup, rather someone's oversight in the implementation (it's essentially ignored now). 

JSONFragment borrows concepts from both Entire and Partial:
* `path()` to provide which file we are managing
* `acl()` to define our scope in the file

So the way to implement a `safe` semantic had been up to one's choice.

So I am ok with the change lets just remove `is_safe()`

File: annet/gen.py [hunk 2]
```diff
@@ -260,7 +260,7 @@ def _old_new_per_device(ctx: OldNewDeviceContext, device: Device, filterer: Filt
         if ctx.args.acl_safe:
             safe_new_files = res.new_files(safe=True)
             safe_new_json_fragment_files = res.new_json_fragment_files(old_json_fragment_files, safe=True)
-            if filters:
+            if len(filters):
```
Comment (azryve) [inline annet/gen.py:263]: lets not ignore pep8 for no reason

https://peps.python.org/pep-0008/#programming-recommendations

```
For sequences, (strings, lists, tuples), use the fact that empty sequences are false:
# Correct:
if not seq:
if seq:
```

**Review by azryve — APPROVED** [review]

**Pull request merged**